gateway/nix/grafbase-gateway.nix

```diff
@@ -0,0 +1,78 @@
+{
+  pkgs,
+  crane,
+  lib,
+  ...
+}: let
+  rustToolchain = pkgs.rust-bin.stable.latest.default;
+  craneLib = (crane.mkLib pkgs).overrideToolchain rustToolchain;
+  workspaceRoot = builtins.path {
+    name = "grafbase-repo-src";
+    path = ../../.;
+  };
+
+  # Aggressively prune the source tree for better caching.
+  extraIgnores = ''
+    *.nix
+    *.md
+    *.sh
+
+    package.json
+    assets.tar.gz
+
+    # We can't ignore /packages wholesale because we need to include
+    # grafbase-sdk/package.json later.
+    /packages/**/*.json
+    /packages/**/*.js
+    /packages/**/*.ts
+    /packages/**/*.html
+
```

Comment:
Would an extraIgnore on `cli/` work here?

Reply:
no because the workspace cargo lock references crates there and the build would fail